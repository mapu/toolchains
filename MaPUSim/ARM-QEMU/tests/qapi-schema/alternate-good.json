# Working example of alternate
{ 'struct': 'Data',
  'data': { '*number': 'int', '*name': 'str' } }
{ 'enum': 'Enum',
  'data': [ 'hello', 'world' ] }
{ 'alternate': 'Alt',
  'data': { 'value': 'int',
            'string': 'Enum',
            'struct': 'Data' } }
