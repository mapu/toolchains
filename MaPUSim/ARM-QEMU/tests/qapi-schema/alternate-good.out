[OrderedDict([('struct', 'Data'), ('data', OrderedDict([('*number', 'int'), ('*name', 'str')]))]),
 OrderedDict([('enum', 'Enum'), ('data', ['hello', 'world'])]),
 OrderedDict([('alternate', 'Alt'), ('data', OrderedDict([('value', 'int'), ('string', 'Enum'), ('struct', 'Data')]))])]
[{'enum_name': 'Enum', 'enum_values': ['hello', 'world']},
 {'enum_name': 'AltKind', 'enum_values': None}]
[OrderedDict([('struct', 'Data'), ('data', OrderedDict([('*number', 'int'), ('*name', 'str')]))])]
